routes/api.js

```diff
@@ -81,5 +82,91 @@ router.post('/station-heartbeat', function(req, res) {
   } else {
     res.sendStatus(401);
   }
-})
+});
+
+//Usage:  Check stations current state.
+//Arguments:
+//         station-id - The id for the station which will be passed in the header of the request
+//Exceptions: 403 Forbidden
+//Description:  This function will look to see what status a station is
+// suppose to be in.  It will then return to the station whether it is suppose
+// to be disabled or enabled.  A response of 200 ok will be returned if all checks
+// are passed or a response of 403 Forbidden will be returned if any check fails.
+router.get('/last-state', function(req, res){
+  var cert = req.socket.getPeerCertificate();
+  var hObject = req.headers;
+  var sid = hObject['station-id'];
+
+  if(sid && cert.subject){
+    db.getStation(sid).then(results => {
+      if(results){
+        if(results['registered'] && cert.subject.CN === results['certCN']){
+          //Store station object
+          let obj = sCache.getMachine(sid);
+
+          //If station was in shared cache return
+          //whether the station was enabled or disabled.
+          if(obj){
+            console.log(obj);
+            if(obj.status === 'enabled'){
+              db.logEvent('station-status request', 'Station: ' + sid + ' enabled and in use by: ' + obj.user).then(function(){
+                res.set({
+                  'Content-Type': 'text/plain',
+                  'station-state': 'enabled',
+                  'user-id-string': obj.user
+                });
+                console.log('obj found + enabled');
+                res.sendStatus(200);
+              });
+            }else{
+              db.logEvent('station-status request', 'Station: ' + sid + ' disabled.').then(function(){
+                res.set({
+                  'Content-Type': 'text/plain',
+                  'station-state': 'disabled',
+                });
+                console.log('obj found + disabled');
+                res.sendStatus(200);
+              });
+            }
+          //If station was not found add it to the stations
+          //shared cache and return status.
+          }else{
+            if(sCache.addMachine(sid)){
+              db.logEvent('station-status request', 'Station: ' + sid + ' disabled and added to the cache.').then(function(){
+                res.set({
+                  'Content-Type': 'text/plain',
+                  'station-state': 'disabled',
+                });
+                console.log('Did we add something to the cache?');
+                res.sendStatus(200);
+              });
+            }else{
+              db.logEvent('station-status request', 'Station: ' + sid + ' has an unknown state and is unable to be add to cache').then(function(){
+                res.sendStatus(403);
```

Comment:
This looks like it's an error handler?
If so, this should return a `500` error.